spread/xunit.go

```diff
@@ -0,0 +1,163 @@
+package spread
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+)
+
+type XUnitTestSuites struct {
+	XMLName xml.Name `xml:"testsuites"`
+	Suites  []*XUnitTestSuite
+}
+
+func (tss *XUnitTestSuites) addSuite(suite *XUnitTestSuite) {
+	tss.Suites = append(tss.Suites, suite)
+}
+
+func (tss *XUnitTestSuites) getSuite(suiteName string) *XUnitTestSuite {
+    for _, s := range tss.Suites {
+        if s.Name == suiteName {
+            return s
+        }
+    }
+	suite := NewXUnitTestSuite(suiteName)
+
+	tss.addSuite(suite)	
+	return suite
+}
+
+type XUnitTestSuite struct {
+	XMLName     xml.Name         `xml:"testsuite"`
+	Successfull int              `xml:"successfull,attr"`
+	Failed      int              `xml:"failed,attr"`
+	Aborted     int              `xml:"aborted,attr"`
+	Total       int              `xml:"total,attr"`
+	Name        string           `xml:"name,attr"`
+	TestCases   []*XUnitTestCase
+}
+
+func NewXUnitTestSuite(suiteName string) *XUnitTestSuite {
+	return &XUnitTestSuite{
+		Successfull: 0,
+		Failed: 0,
+		Aborted: 0,
+		Total: 0,
+		Name: suiteName,
+		TestCases: []*XUnitTestCase{},
+	}
+}
+
+func (ts *XUnitTestSuite) addTest(test *XUnitTestCase) {
```

Comment:
looks very much JSONUnitTestCases, maybe you could have something like genericUnitTestsReporter which then can be embedded in respective variant?

Reply:
done